### PR TITLE
Lab1 + Lab2 implemented

### PR DESCRIPTION
**Pull request by Tasadan-Filip**

### CODE REVIEW

File: Lab_2/Lexic.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
+Alphabet:
+
+a. Upper (A-Z) and lower case letters (a-z) of the English alphabet
+
+       b. Underline character '_';
```
Comment (Daniel-Todasca) [inline Lab_2/Lexic.txt]: Underline character is not used below - add an usage or remove from the alphabet

File: Lab_2/Lexic.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    zero = "0"
+    positiveDigit = "1" |...| "9"
+
+      c.constants
+
+1.integer - rule:
+
+      noconst="+"positiveNo|"-"positiveNo|no 
+
+      no:=zero|digit|positiveDigit{no}
```
Comment (Daniel-Todasca) [inline Lab_2/Lexic.txt]: Redundant recursivity: zero|digit
Left-handside is used recursively in right-handside
no := digit|positiveDigit{digit}
Also, EBNF does not use :=

File: Lab_2/Lexic.txt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      positiveNo = positiveDigit{digit}
+
+2.character
+
+    character='letter'|'digit'
+
+3.string
+
+      constchar:="string"
```
Comment (Daniel-Todasca) [inline Lab_2/Lexic.txt]: Use emptystring here
constchar := "string" | emptyString

File: Lab_2/Syntax.in [hunk 4]
```diff
@@ -0,0 +1,43 @@
+Sintactical rules: 
+
+program = decllist ";" stmtlist
```
Comment (Daniel-Todasca) [inline Lab_2/Syntax.in:3]: declarations only available at the beginning of the program?

File: Lab_2/Syntax.in [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+stmtlist = stmt | stmt ";" stmtlist
+
+stmt = simplstmt | structstmt
+
+simplstmt = assignstmt | iostmt
+
+assignstmt = identifier "=" expression
+
+expression = expression ("+"|"-") term | term|constant
```
Comment (Daniel-Todasca) [inline Lab_2/Syntax.in]: Redundant term:
expression = expression ("+"|"-") term | term (constant appears below)

**Review by Daniel-Todasca — CHANGES_REQUESTED** [review]
remove reduntant recursivity

File: Lab_2/Syntax.in (hunk 4 above)
Comment (Tasadan-Filip) [reply Lab_2/Syntax.in:3]: Yes, this is how I want the program to work

**Pull request merged**